lib/impl/AzureBlobFiles.js

```diff
@@ -278,13 +282,36 @@ class AzureBlobFiles extends Files {
    */
   async _addAccessPolicyIfNotExists () {
     const identifier = await this._getAccessPolicy()
-    logger.debug('found access policy with identifier ' + identifier)
-    if (!identifier) {
+    if (identifier) {
+      logger.debug('found access policy with identifier ' + identifier)
+      if (this.hasOwnCredentials) {
+        // check if identifier is custom or not
+        if (this._hasCustomAccessPolicies || this._isCustomPolicy(identifier)) {
+          const msg = 'Container has one or more custom policies defined. Either remove all custom policies or use another container.'
+          logAndThrow(new codes.ERROR_INIT_FAILURE({ messageValues: [msg], sdkDetails: {} }))
+        }
+      }
+    } else {
       logger.debug('adding default access policy')
       await this._setAccessPolicy()
     }
   }
 
+  /**
+   * [Internal] Check if identifier is custom or not
+   *
+   * @private
+   */
+  _isCustomPolicy (identifier) {
+    if (identifier.startsWith('aio-lib-files')) {
+      return false
+    } else {
+      // if its uuidv4 type of identifier then consider it as non custom, this is to support any already created policies by aio-lib-flies
+      const testUUIDv4 = /^[0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{12}$/gi
```

Comment:
I think we should only use that condition to avoid weird corner cases with old vs new ids

Meaning isCustom equals true only and only if it's not a uuidv4 and ditch the `aio-lib-files` prefix

Reply:
I feel it will be good if we add the prefix, that clearly helps to differentiate between any policy lib creates vs anything else. The current two conditions of prefix check and uuidv4 format should cover any cases relevant to our usecase.

Reply:
On second thought maybe since we are enforcing a single policy we won't really need a prefix to differenciate. Will remove the logic to add it.